source/AMDSClientUi.cpp

```diff
@@ -0,0 +1,332 @@
+		return;
+	}
+
+	QStringList selectedBufferNames;
+	foreach (const QModelIndex &index, bufferNameListView_->selectionModel()->selectedIndexes())
+	{
+		selectedBufferNames.append(index.data(Qt::DisplayRole).toString());
+	}
+
+	QString hostName = server->hostName();
+	quint16 portNumber = server->portNumber();
+	quint8 requestTypeId = (quint8)requestTypeComboBox_->currentIndex();
+	QString bufferName = selectedBufferNames.value(0);
+	QDateTime time1 = time1Edit_->dateTime();
+	QDateTime time2 = time2Edit_->dateTime();
+	QString value1 = count1Edit_->text();
+	QString value2 = count2Edit_->text();
+	QString continuousSocket = activeContinuousConnectionComboBox_->currentText();
+
+	bool includeStatus = includeStatusDataCheckbox_->isChecked();
+	bool enableFlattening = enableFlattenDataCheckbox_->isChecked();
+
+	AMDSClientRequestDefinitions::RequestType clientRequestType = (AMDSClientRequestDefinitions::RequestType)requestTypeId;
+	if (	(selectedBufferNames.count() == 0)
+		 && (clientRequestType != AMDSClientRequestDefinitions::Continuous )
+		 && (clientRequestType != AMDSClientRequestDefinitions::Statistics)) {
+		QMessageBox::information( this, "AMDS Client Example", "Didn't select any buffer(s)!");
+		return;
+	}
+
+	switch(clientRequestType) {
+	case AMDSClientRequestDefinitions::Introspection:
+		clientAppController_->requestClientData(hostName, portNumber, bufferName);
+		break;
+	case AMDSClientRequestDefinitions::Statistics:
+		clientAppController_->requestClientData(hostName, portNumber);
+		break;
+	case AMDSClientRequestDefinitions::StartTimePlusCount:
+		clientAppController_->requestClientData(hostName, portNumber, bufferName, time1, value1.toInt(), includeStatus, enableFlattening);
+		break;
+	case AMDSClientRequestDefinitions::RelativeCountPlusCount:
+		clientAppController_->requestClientData(hostName, portNumber, bufferName, value1.toInt(), value2.toInt(), includeStatus, enableFlattening);
+		break;
+	case AMDSClientRequestDefinitions::StartTimeToEndTime:
+		clientAppController_->requestClientData(hostName, portNumber, bufferName, time1, time2, includeStatus, enableFlattening);
+		break;
+	case AMDSClientRequestDefinitions::MiddleTimePlusCountBeforeAndAfter:
+		clientAppController_->requestClientData(hostName, portNumber, bufferName, time1, value1.toInt(), value2.toInt(), includeStatus, enableFlattening);
+		break;
+	case AMDSClientRequestDefinitions::Continuous:
+		clientAppController_->requestClientData(hostName, portNumber, selectedBufferNames, value1.toInt(), includeStatus, enableFlattening, continuousSocket);
+		if (continuousSocket.length() > 0) {
+			AMDSErrorMon::information(this, 0, QString("Hand shake message: %1").arg(continuousSocket));
+		}
+		break;
+	default:
+		AMDSErrorMon::alert(this, 0, QString("Invalide client request type: %1").arg(clientRequestType));
+	}
+}
+
+void AMDSClientUi::resetBufferListView(QStringList &bufferNames)
+{
+	QStringListModel *bufferNamesModel = new QStringListModel(this);
```

Comment:
Does QListView handle the memory cleanup of this QStringListModel?  If not, you'll need to handle this as it is a memory leak.